### PR TITLE
CFY-4639-AWS-based-offline-package-tests-missing-functionality

### PR DESCRIPTION
**Pull request by max-orlov**

### CODE REVIEW

File: system_tests/resources/start-ec2-worker-vm-blueprint.yaml [hunk 1]
```diff
@@ -60,12 +64,23 @@ node_templates:
...
+    type: cloudify.aws.nodes.WindowsInstance
     properties:
       install_agent: false
       image_id: { get_input: image_id }
       instance_type: { get_input: instance_type }
       aws_config: *AWS_CONFIG
+      parameters:
+        user_data: |
+          <powershell>
+          &winrm quickconfig -q
```
Comment (ran-z) [inline system_tests/resources/start-ec2-worker-vm-blueprint.yaml]: are all of those really necessary?
talk with NirB, im pretty sure he's got a test where only one of these lines is used

File: system_tests/resources/test-ec2-fileserver-vm-blueprint.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      - type: cloudify.aws.relationships.instance_connected_to_security_group
+        target: security_group
+
+outputs:
+  vm_public_ip_address:
+    description: Public IP of the worker VM created for purposes of a test
+    value: { get_attribute: [elastic_ip, aws_resource_id] }
+  vm_private_ip_address:
+    value: { get_attribute: [server, ip]}
+  windows_vm_password:
```
Comment (ran-z) [inline system_tests/resources/test-ec2-fileserver-vm-blueprint.yaml]: is this necessary? couldnt you have taken it from the windows host node instance? (it should be saved on it as a runtime property)

File: system_tests/resources/start-ec2-worker-vm-blueprint.yaml (hunk 1 above)
Comment (max-orlov) [inline system_tests/resources/start-ec2-worker-vm-blueprint.yaml]: Well, probably we could loose some (I just copied it from this other blueprint - better safe than sorry). However, it appears that we need at least 2 of them (the 5985 line obviously and maybe the basic=true or the allowunencrypted), since I've tried it with only the 5985 and it says there's a problem authenticating with simple password on a different domain in windows - I've already talked with nirbir about it.

File: system_tests/resources/test-ec2-fileserver-vm-blueprint.yaml (hunk 2 above)
Comment (max-orlov) [inline system_tests/resources/test-ec2-fileserver-vm-blueprint.yaml]: You are right, i haven't really checked the part where get to know the password for the win machine (Nirbir added it).

Comment (max-orlov) [inline system_tests/resources/test-ec2-fileserver-vm-blueprint.yaml]: I think this part definitely doesn't belong here. I think this is a misplaced copy+paste.

Comment (nirbir) [inline system_tests/resources/test-ec2-fileserver-vm-blueprint.yaml]: Yes, this needs to be removed. The server is not a windows instance here

File: system_tests/resources/test-windows-bootstrap-blueprint.yaml [hunk 3]
```diff
@@ -1,136 +0,0 @@
-
```
Comment (nirbir) [inline system_tests/resources/test-windows-bootstrap-blueprint.yaml]: Why was this removed?

File: system_tests/test_cli_package.py [hunk 4]
```diff
@@ -183,26 +194,23 @@ def _execute_command(self, cmd, within_cfy_env=False, sudo=False,
...
-    def install_cli(self):
+    def install_cli(self, *args, **kwargs):
         self.logger.info('installing cli...')
 
-        self._get_resource(self.cli_package_url, ops='-LO', sudo=True)
-        self._get_resource('https://bootstrap.pypa.io/get-pip.py', ops='-L',
-                           pipe_command='sudo python2.7 -')
+        self._get_resource(self.cli_package_url, curl_ops='-LO', sudo=True)
+        self._get_resource('https://bootstrap.pypa.io/get-pip.py',
+                           curl_ops='-L | sudo python2.7 -')
```
Comment (nirbir) [inline system_tests/test_cli_package.py]: Why was the pipe_command removed? It seems clearer with this var

File: system_tests/test_cli_package.py [hunk 5]
```diff
@@ -241,54 +241,41 @@ def bootstrap_manager(self):
         if self.is_install_plugins():
             install_plugins = '--install-plugins'
         out = self._execute_command(
-            'bootstrap -p {0} -i {1} {2}'.format(
-                self.test_manager_blueprint_path,
-                self.remote_bootstrap_inputs_path,
+            """bootstrap -p {0} -i "{1}" {2}""".format(
+                self.manager_blueprint_path,
+                json.dumps(inputs).replace('"', "'"),
```
Comment (nirbir) [inline system_tests/test_cli_package.py]: AFAIK the json.dumps string is different from our required inputs format of "key1=value1;key2=value2"

File: system_tests/test_cli_package.py [hunk 6]
```diff
@@ -241,54 +241,41 @@ def bootstrap_manager(self):
...
         deployment_id = 'deployment-{0}'.format(uuid.uuid4())
-        self.prepare_deployment()
 
         self.logger.info('Creating deployment: {0}'.format(deployment_id))
         self._execute_command(
-            'deployments create -b {0} -d {1} -i {2}'
+            """deployments create -b {0} -d {1} -i "{2}" """
             .format(blueprint_id, deployment_id,
-                    self.remote_deployment_inputs_path),
+                    json.dumps(self.deployment_inputs).replace('"', "'")),
```
Comment (nirbir) [inline system_tests/test_cli_package.py]: AFAIK the json.dumps string is different from our required inputs format of "key1=value1;key2=value2"

File: system_tests/resources/start-ec2-worker-vm-blueprint.yaml [hunk 7]
```diff
@@ -80,3 +95,5 @@ outputs:
     value: { get_attribute: [elastic_ip, aws_resource_id] }
   vm_private_ip_address:
     value: { get_attribute: [server, ip]}
+  windows_vm_password:
```
Comment (ran-z) [inline system_tests/resources/start-ec2-worker-vm-blueprint.yaml]: should this really be an output?

File: system_tests/test_windows_bootstrap.py [hunk 8]
```diff
@@ -213,31 +278,22 @@ class TestWindowsOfflineBootstrap(TestWindowsBase, TestOfflineCliPackage):
...
-        self._execute_command(wget_cmd)
+    def prepare_cli(self):
+        self.logger.info('Downloading Windows CLI package from: {0}'
+                         .format(self.cli_package_url))
+        self._get_resource(self.cli_package_url, CLI_PACKAGE_EXE)
 
         return CLI_PACKAGE_EXE
 
-    def _install_cli(self, package_name):
+    def install_cli(self, package_name):
```
Comment (nirbir) [inline system_tests/test_windows_bootstrap.py]: Implemented in super class windowsbase

File: system_tests/resources/test-ec2-fileserver-vm-blueprint.yaml (hunk 2 above)
Comment (ran-z) [inline system_tests/resources/test-ec2-fileserver-vm-blueprint.yaml]: @nirbir i meant so in general - there's also the other blueprint where the windows VM password appears in the outputs, and there it really is a Win VM - yet i dont see why we shouldnt just use the value from the runtime properties instead..

File: system_tests/test_cli_package.py [hunk 9]
```diff
@@ -183,26 +194,23 @@ def _execute_command(self, cmd, within_cfy_env=False, sudo=False,
...
-                      pipe_command=''):
-        if pipe_command:
-            pipe_command = "| {0}".format(pipe_command)
+    def _get_resource(self, resource_address, sudo=False, curl_ops='',
+                      *args, **kwargs):
 
-        return self._execute_command('curl {0} {1} {2}'
-                                     .format(ops, resource_address,
-                                             pipe_command),
+        return self._execute_command('curl {0} {1}'
```
Comment (ran-z) [inline system_tests/test_cli_package.py]: for future reference, "ops" = "operations"; "opts" = "options" :P

File: system_tests/resources/start-ec2-worker-vm-blueprint.yaml (hunk 1 above)
Comment (nirbir) [inline system_tests/resources/start-ec2-worker-vm-blueprint.yaml]: It's true that I need only one of those lines, but I never user userdata. I ran the line manually on the client, after it was up. Maybe when using userdata more commands are required.

File: system_tests/test_cli_package.py [hunk 10]
```diff
@@ -241,54 +241,41 @@ def bootstrap_manager(self):
         if self.is_install_plugins():
             install_plugins = '--install-plugins'
         out = self._execute_command(
-            'bootstrap -p {0} -i {1} {2}'.format(
-                self.test_manager_blueprint_path,
-                self.remote_bootstrap_inputs_path,
+            """bootstrap -p {0} -i "{1}" {2}""".format(
```
Comment (ran-z) [inline system_tests/test_cli_package.py]: also for future reference, triple quotes is for multiline strings. you should probably just alternate between regular quotes and single quotes (" and ') or use escaping here.

also note that you're passing shell arguments here, so it'd make more sense for {1} to be surrounded with single quotes ('), since that tells the shell to take the string as-is, as opposed to quotes (") which the shell might interpret differently.

File: system_tests/test_windows_bootstrap.py [hunk 11]
```diff
@@ -12,129 +12,225 @@
...
+            'aws_access_key_id': self.env.aws_access_key_id,
+            'aws_secret_access_key': self.env.aws_secret_access_key,
+            'ec2_region_name': self.env.ec2_region_name,
+            'manager_keypair_name': '{0}-manager-keypair'.format(self.prefix),
+            'agent_keypair_name': '{0}-agent-keypair'.format(self.prefix),
+            'ssh_user': 'centos',
+            'agents_user': 'centos',
+            'image_id': self.env.centos_7_image_id,
+            'instance_type': self.env.medium_instance_type,
+            # TODO: by default the values go to ..\Administrator\.ssh\<kp_name>. since '.ssh' never really exists this causes an error. This is probably not the wanted behaviour.  # NOQA
```
Comment (ran-z) [inline system_tests/test_windows_bootstrap.py]: ..?
who's TODOing this :)

File: system_tests/test_windows_bootstrap.py [hunk 12]
```diff
@@ -213,31 +278,22 @@ class TestWindowsOfflineBootstrap(TestWindowsBase, TestOfflineCliPackage):
...
+    def install_cli(self, package_name):
         self.logger.info('Installing CLI...')
-        with self.dns():
-            self._execute_command(
-                '.\{0} /SILENT /VERYSILENT /SUPPRESSMSGBOXES /DIR="{1}"'
-                ''.format(package_name, self.client_cfy_work_dir))  # NOQA
-            self.logger.info('Verifying CLI installation...')
-            self._execute_command('--version', within_cfy_env=True)
+        self._execute_command(
+            '.\{0} /SILENT /VERYSILENT /SUPPRESSMSGBOXES /DIR="{1}"'
```
Comment (ran-z) [inline system_tests/test_windows_bootstrap.py]: VERYSILENT

File: system_tests/resources/test-ec2-fileserver-vm-blueprint.yaml (hunk 2 above)
Comment (nirbir) [inline system_tests/resources/test-ec2-fileserver-vm-blueprint.yaml]: The reason is that I knew how to access the password from an output, and didn't know how to access runtime properties from the test. Can be changed.

File: system_tests/test_cli_package.py (hunk 10 above)
Comment (max-orlov) [inline system_tests/test_cli_package.py]: Yeah, so it was a single quote, but windows is being lame and doesn't get it as it should...
And the triple quotes is just for readability ( I could've used the escape char for the inner double quotes).

File: system_tests/test_windows_bootstrap.py (hunk 11 above)
Comment (max-orlov) [inline system_tests/test_windows_bootstrap.py]: Oh, totally forgot about it. good i wrote it down.

File: system_tests/test_windows_bootstrap.py (hunk 12 above)
Comment (max-orlov) [inline system_tests/test_windows_bootstrap.py]: yeah, that one made me laugh too.

File: system_tests/test_windows_bootstrap.py (hunk 8 above)
Comment (max-orlov) [inline system_tests/test_windows_bootstrap.py]: Yeah, so the windowsbase version would work only in an online env. So it should probably move to the online test class.

File: system_tests/test_cli_package.py (hunk 10 above)
Comment (ran-z) [inline system_tests/test_cli_package.py]: hmm ok i get it
guess this order of quotes should work fine on linux too

File: system_tests/test_windows_bootstrap.py (hunk 8 above)
Comment (nirbir) [inline system_tests/test_windows_bootstrap.py]: Actually, the online windows test has an install_cli method, and so does the offline. windowsbase doesn't have it, so it seems fine

File: system_tests/test_cli_package.py [hunk 13]
```diff
@@ -211,84 +239,106 @@ def add_dns_nameservers_to_manager_blueprint(self, local_modify_script):
         self.logger.info(
             'Uploading {0} to {1} on manager...'.format(local_modify_script,
                                                         remote_modify_script))
-        fab.put(local_modify_script, remote_modify_script, use_sudo=True)
-        self.logger.info(
-            'Adding DNS name servers to remote manager blueprint...')
-        fab.run('sudo python {0} {1}'.format(
-            remote_modify_script, self.test_manager_blueprint_path))
+        with fab_env(**self.centos_client_env):
```
Comment (nirbir) [inline system_tests/test_cli_package.py]: default fab env? safe to assume centos?

File: system_tests/test_cli_package.py [hunk 14]
```diff
@@ -211,84 +239,106 @@ def add_dns_nameservers_to_manager_blueprint(self, local_modify_script):
...
-
-    def create_deployment(self, blueprint_id):
+    def create_deployment(self, blueprint_id, *_):
         deployment_id = 'deployment-{0}'.format(uuid.uuid4())
-        self.prepare_deployment()
 
         self.logger.info('Creating deployment: {0}'.format(deployment_id))
-        self._execute_command(
-            'deployments create -b {0} -d {1} -i {2}'
+        self._execute_command_on_linux(
```
Comment (nirbir) [inline system_tests/test_cli_package.py]: why on linux?

File: system_tests/test_cli_package.py [hunk 15]
```diff
@@ -298,14 +348,19 @@ def install_deployment(self, deployment_id):
             'Waiting for 15 seconds before installing deployment...')
         time.sleep(15)
         self.logger.info('Installing deployment...')
-        self._execute_command('executions start -d {0} -w install'
-                              .format(deployment_id),
-                              within_cfy_env=True, retries=2)
+        self._execute_command_on_linux(
```
Comment (nirbir) [inline system_tests/test_cli_package.py]: executor

File: system_tests/test_cli_package.py [hunk 16]
```diff
@@ -298,14 +348,19 @@ def install_deployment(self, deployment_id):
...
+            'executions start -d {0} -w install'
+            .format(deployment_id),
+            self.centos_client_env,
+            within_cfy_env=True, retries=2)
 
     def uninstall_deployment(self):
         self.logger.info('Uninstalling deployment...')
-        self._execute_command('executions start -d {0} -w uninstall'
-                              .format(self.deployment_id), within_cfy_env=True)
+        self._execute_command_on_linux(
```
Comment (nirbir) [inline system_tests/test_cli_package.py]: executor

File: system_tests/test_cli_package.py [hunk 17]
```diff
@@ -349,53 +406,70 @@ def cleanup(self):
 
     def teardown_manager(self):
         self.logger.info('Tearing down Cloudify manager...')
-        self._execute_command('teardown -f --ignore-deployments',
-                              within_cfy_env=True)
+        self._execute_command_on_linux(
```
Comment (nirbir) [inline system_tests/test_cli_package.py]: why does this work??

File: system_tests/test_cli_package.py [hunk 18]
```diff
@@ -404,29 +478,36 @@ def install_python27(self):
...
-            self._execute_command('yum groupinstall -y \'development '
-                                  'tools\'', sudo=True)
-            self._execute_command('yum install -y zlib-devel bzip2-devel '
-                                  'openssl-devel xz-libs', sudo=True)
-            self._execute_command('curl -LO http://www.python.org/ftp/python/'
-                                  '2.7.8/Python-2.7.8.tar.xz', sudo=True)
-            self._execute_command('xz -d Python-2.7.8.tar.xz', sudo=True)
-            self._execute_command('tar -xvf Python-2.7.8.tar', sudo=True)
+            self._execute_command_on_linux('yum -y update', sudo=True)
+            self._execute_command_on_linux(
```
Comment (nirbir) [inline system_tests/test_cli_package.py]: why does this work on windows?

File: system_tests/test_offline_cli_package.py [hunk 19]
```diff
@@ -38,83 +40,107 @@ def _test_cli_package(self):
...
+        tmp_manager_blueprint_dir = tempfile.mkdtemp()
+        mng_blueprint_tmp_path = os.path.join(
+            tmp_manager_blueprint_dir,
+            'mng_bluprint.yaml')
+        copyfile(self.env._manager_blueprint_path,
+                 mng_blueprint_tmp_path)
+
+        # removing dns
+        user_data = """#!/bin/bash
+mkdir /tmp/nir
```
Comment (nirbir) [inline system_tests/test_offline_cli_package.py]: delete

File: system_tests/test_offline_cli_package.py [hunk 20]
```diff
@@ -123,62 +149,44 @@ def _update_example_sg(self):
                                           self.branch),
                          self.app_blueprint_file)
         example_blueprint = StringIO()
-        fab.get(example_blueprint_path, example_blueprint)
-        hello_world_blueprint_yaml = yaml.load(example_blueprint.getvalue())
+        with fab_env(**self.centos_client_env):
```
Comment (nirbir) [inline system_tests/test_offline_cli_package.py]: windows env?

File: system_tests/test_windows_bootstrap.py (hunk 8 above)
Comment (nirbir) [inline system_tests/test_windows_bootstrap.py]: move to windows base?

File: system_tests/test_cli_package.py [hunk 21]
```diff
@@ -58,7 +58,7 @@ def client_cfy_work_dir(self):
 
     @property
     def client_user(self):
-        raise NotImplementedError
+        return 'centos'
```
Comment (nirbir) [inline system_tests/test_cli_package.py]: should this stay here? meeseeks

File: system_tests/test_cli_package.py [hunk 22]
```diff
@@ -211,84 +239,106 @@ def add_dns_nameservers_to_manager_blueprint(self, local_modify_script):
...
-                         .format(hello_world_url, blueprint_id))
-        self._execute_command('blueprints publish-archive '
-                              '-l {0} -n {1} -b {2}'
-                              .format(hello_world_url,
-                                      self.app_blueprint_file,
-                                      blueprint_id),
-                              within_cfy_env=True)
+                         .format(source_archive, blueprint_id))
+        if not executor:
+            executor = self._execute_command_on_linux
```
Comment (nirbir) [inline system_tests/test_cli_package.py]: client executor?

File: system_tests/test_offline_cli_package.py [hunk 23]
```diff
@@ -38,147 +38,131 @@ def _test_cli_package(self):
...
             self.addCleanup(self.uninstall_deployment)
             self.install_deployment(self.deployment_id)
             self.assert_deployment_working(
                 self._get_app_property('http_endpoint'))
 
-    def _update_hosts_file(self, resolution):
+    def prepare_manager_blueprint(self):
+        super(TestOfflineCliPackage, self).prepare_manager_blueprint()
+
+        tmp_manager_blueprint_dir = tempfile.mkdtemp()
```
Comment (nirbir) [inline system_tests/test_offline_cli_package.py]: maybe take the manager bp from the package client

File: system_tests/test_offline_cli_package.py [hunk 24]
```diff
@@ -38,147 +38,131 @@ def _test_cli_package(self):
...
-
-        # Update the inputs to include the additional inputs
-        inputs.update(additional_inputs)
-        inputs_file.seek(0)
-        json.dump(inputs, inputs_file)
-        inputs_file.truncate()
-
-        # Write the inputs back to the cli vm.
-        fab.put(inputs_file, self.remote_bootstrap_inputs_path)
+        self._execute_command_on_linux(
```
Comment (nirbir) [inline system_tests/test_offline_cli_package.py]: client executor?

File: system_tests/test_cli_package.py [hunk 25]
```diff
@@ -211,84 +248,105 @@ def add_dns_nameservers_to_manager_blueprint(self, local_modify_script):
...
-        self.manager_blueprints_repo_dir = '{0}/cloudify-manager-blueprints/' \
-                                           .format(self.client_cfy_work_dir)
-        self.test_manager_blueprint_path = \
-            os.path.join(self.manager_blueprints_repo_dir,
+        self.logger.info('Preparing manager blueprint')
+        self.manager_blueprints_dir = \
+            '{0}/cloudify-manager-blueprints'.format(
+                self.client_cfy_work_dir)
+        self.manager_blueprint_path = \
+            os.path.join(self.manager_blueprints_dir,
```
Comment (max-orlov) [inline system_tests/test_cli_package.py]: check how prepare_manager_blueprint is implemented

File: system_tests/test_cli_package.py [hunk 26]
```diff
@@ -211,84 +248,105 @@ def add_dns_nameservers_to_manager_blueprint(self, local_modify_script):
...
+        self.assertIn('bootstrapping complete', out, 'Bootstrap has failed')
 
         self.manager_ip = self._manager_ip()
         self.client = CloudifyClient(self.manager_ip)
         self.addCleanup(self.teardown_manager)
 
-    def get_hello_world_url(self):
-        return EXAMPLE_URL.format(HELLO_WORLD_EXAMPLE_NAME, self.branch)
+    def get_example(self, example_url):
+        return example_url
```
Comment (max-orlov) [inline system_tests/test_cli_package.py]: check if needed

File: system_tests/test_cli_package.py [hunk 27]
```diff
@@ -298,23 +356,30 @@ def install_deployment(self, deployment_id):
...
         self.install_cli()
         self.prepare_manager_blueprint()
         self.add_dns_nameservers_to_manager_blueprint(
             os.path.join(os.path.dirname(__file__),
                          'resources/add_nameservers_to_subnet.py'))
-        self.bootstrap_manager()
-        blueprint_id = self.publish_hello_world_blueprint()
+        self.bootstrap_manager(self.bootstrap_inputs)
+        example_path = self.get_example(self.helloworld_url)
+        blueprint_id = self.publish_hello_world_blueprint(example_path)
```
Comment (max-orlov) [inline system_tests/test_cli_package.py]: remove the example_path

File: system_tests/test_cli_package.py [hunk 28]
```diff
@@ -322,12 +387,13 @@ def _test_cli_package(self):
...
     def _manager_ip(self):
-        return self._execute_command(
+        return self._execute_command_on_linux(
             'source {0}/env/bin/activate && {1}'.format(
                 self.client_cfy_work_dir,
                 'python -c "from cloudify_cli import utils;'
                 'print utils.get_management_server_ip()"'
-            )
+            ),
+            self.centos_client_env
```
Comment (max-orlov) [inline system_tests/test_cli_package.py]: fabric_env=

File: system_tests/test_cli_package.py [hunk 29]
```diff
@@ -444,15 +543,22 @@ def wait_for_vm_to_become_ssh_available(env_settings, executor,
     local_env_setting = copy.deepcopy(env_settings)
     local_env_setting.update({'abort_exception': FabException})
     local_env_setting.update({'timeout': timeout})
+
+    # Sometimes when not enough time is given before logging in, the iaas
+    # requests for some password.
+    # cooldown_time = 60
+    # logger.info('Waiting for {0} seconds before checking ssh connection'
+    #             .format(cooldown_time))
+    time.sleep(10)
```
Comment (max-orlov) [inline system_tests/test_cli_package.py]: remove

File: system_tests/test_offline_cli_package.py [hunk 30]
```diff
@@ -38,147 +35,90 @@ def _test_cli_package(self):
...
+    def prepare_hosts_file(self, iaas_mapping, fabric_env):
+        self.update_hosts_file(iaas_mapping, fabric_env)
+
+    def update_hosts_file(self, resolution, fabric_env):
         """
         This function is needed in order to make the transition to windows
         easier
         :param resolution: The resolution rule of url per address.
         :return:
         """
```
Comment (max-orlov) [inline system_tests/test_offline_cli_package.py]: remove - awkward.

File: system_tests/test_offline_cli_package.py [hunk 31]
```diff
@@ -38,147 +35,90 @@ def _test_cli_package(self):
...
+        return yaml_file
+
+    def _get_remote_blueprint(self, env=None):
+        env = env or self.centos_client_env
+        manager_blueprint = StringIO()
+        with fab_env(**env):
+            fab.get(self.manager_blueprint_path, manager_blueprint)
+        return yaml.load(manager_blueprint.getvalue())
+
+    def prepare_hosts_file(self, iaas_mapping, fabric_env):
```
Comment (max-orlov) [inline system_tests/test_offline_cli_package.py]: name is not clear and shares functionality with update_hosts_file

File: system_tests/test_offline_cli_package.py [hunk 32]
```diff
@@ -442,8 +375,15 @@ def get_processed_inputs(self):
         """
         return self.processed_inputs
 
-    def _execute_command(self, cmd, sudo=False, pty=True, log_cmd=True,
-                         retries=0, warn_only=False):
+    def _execute_command(
+            self,
+            cmd,
+            fabric_env=None,
```
Comment (max-orlov) [inline system_tests/test_offline_cli_package.py]: maybe this is not needed

File: system_tests/test_windows_bootstrap.py [hunk 33]
```diff
@@ -147,65 +334,54 @@ def dns(self, dns_name_servers=('8.8.8.8', '8.8.4.4')):
...
-            'Downloading Windows CLI package from: {0}'.format(
-                self.cli_package_url))
-        self._execute_command(wget_cmd)
-        self.logger.info('Installing CLI...')
-        self._execute_command(
-            '.\{0} /SILENT /VERYSILENT /SUPPRESSMSGBOXES /DIR="{1}"'
-            ''.format(CLI_PACKAGE_EXE, self.cfy_work_dir))  # NOQA
-        self.logger.info('Verifying CLI installation...')
-        self._execute_command('--version', within_cfy_env=True)
+    def prepare_manager_blueprint(self):
```
Comment (max-orlov) [inline system_tests/test_windows_bootstrap.py]: maybe this isn't needed
